src/Microsoft.AspNetCore.Razor.Language/RazorCodeGenerationOptionsBuilder.cs

```diff
@@ -5,7 +5,9 @@ namespace Microsoft.AspNetCore.Razor.Language
 {
     public abstract class RazorCodeGenerationOptionsBuilder
     {
-        public abstract bool DesignTime { get; }
+        public virtual RazorConfiguration Configuration => null;
+
+        public abstract bool DesignTime { get; set; }
```

Comment:
@Eilon - how do you feel about a small breaking change like this?? The alternative would be to make the setter a method call, which would be gross but non-breaking.

This is a class that someone is very unlikely to have inherited from.

Reply:
The other Idea that I have is to add `IsDesignTime` and obsolete this one. I don't like that idea as much because there's a bunch of other places with a `DesignTime` property and this one would stick out.

Reply:
It does appear that no one in the universe at all is using this API outside of us. Nothing on the WORLD-WIDE web, nothing on GitHub.

I'd still be more comfortable if we didn't break it, though. I'd like us to be more serious about not breaking things whenever reasonably possible. Adding a method for setting is a bit gross, but hardly a big deal for us to do.

Reply:
K